File: chat-service/src/main/java/org/example/domain/chat/domain/ChatMessage.java
package org.example.domain.chat.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChatMessage {
    
    public enum MessageType {
        ENTER, TALK, LEAVE
    }
    
    private MessageType type;
    private String roomId;
    private String sender;
    private String message;
    private String timestamp;

    public String getPrefixMessage(){
        return this.sender + ":" + this.message;
    }

}

File: chat-server-kotiln/src/main/kotlin/com/chat/chatserverkotiln/domain/chat/domain/ChatMessage.kt
package com.chat.chatserverkotiln.domain.chat.domain

data class ChatMessage(
    val type: MessageType, // "CHAT", "JOIN", "LEAVE", "SYSTEM"
    val roomId: String,
    val userId: String?,
    val content: String,
    val timestamp: Long = System.currentTimeMillis()
)


enum class MessageType {
    CHAT,
    JOIN,
    LEAVE,
    SYSTEM
}